enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,483 @@
+Operator targets the "master" pool rather than the "worker" pool whenever the
+"Infrastructure Topology" is set to "SingleReplica". This will ensure that the
+`router-default` deployment created by the Cluster Ingress Operator will always
+run on the single control plane node, and as a result any
+`*.apps.<cluster>.<base>` DNS entries which originally pointed at the single
+control plane node will remain correct even in the face of newly added worker
+nodes.
+
+Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+This enhancement does not modify/add any API
+
+### Implementation Details/Notes/Constraints
+
+#### Infrastructure Topology value
+
+The following table tries to cover the various installation scenarios, the
+existing values for the "Infrastrcture Topology", the target pool in the
+default `IngressController` and the new proposed values for those two parameters
+under those scenarios.
+
+The following abbreviations are used to get a more compact table -
+
+- `D1W`/`D2W` = The amount of workers during installation / The amount of workers added post-installation
+
+- `SR`/`HA` = SingleReplica / HighlyAvailable
+
+- `CIT`/`PIT` = Current "Infrastrcture Topology" / Proposed "Infrastrcture Topology"
+
+- `M`/`W` = Master / Worker
+
+- `CTP`/`PTP` = Current `IngressController` target pool / Proposed `IngressController` target pool
+
+- `SNO` = Single control-plane node cluster
+
+|Scenario                                |Platform|D1W|D2W|CIT|PIT|CTP|PTP|Is a load balancer currently needed?             |Will a load balancer be needed after proposal?|Current Support                     |
+|----------------------------------------|--------|---|---|---|---|---|---|-------------------------------------------------|----------------------------------------------|------------------------------------|
+|None platform SNO no workers            |None    |0  |0  |SR |SR |W  |M  |No                                               |No                                            |Works, supported, tested            |
+|Cloud platform SNO no workers           |Cloud   |0  |0  |SR |SR |W  |M  |No                                               |No                                            |Works, unsupported, tested          |
+|None platform SNO day1 single worker    |None    |1  |1  |-  |SR |-  |M  |Yes, ingress floats                              |No, ingress pinned to master pool             |Impossible to install               |
```

Comment:
I am missing what the limitation is that makes a SNO cluster with day-1 workers an impossibility for the "none" platform. I see mention above about the worker ignition file not being generated when doing a bootstrap-in-place installation. Is that what the "impossibility" means here? If so, then that assume thats (1) the SNO installation is a bootstrap-in-place installation and (2) that the user does not know how to run `openshift-install create ignition-configs` to generate the worker ignition.

Reply:
Will respond on the duplicate comment